dali/kernels/signal/window/extract_windows_cpu.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_SIGNAL_WINDOW_EXTRACT_WINDOWS_CPU_H_
+#define DALI_KERNELS_SIGNAL_WINDOW_EXTRACT_WINDOWS_CPU_H_
+
+#include <memory>
+#include <complex>
+#include "dali/core/common.h"
+#include "dali/core/error_handling.h"
+#include "dali/core/format.h"
+#include "dali/core/util.h"
+#include "dali/kernels/kernel.h"
+
+namespace dali {
+namespace kernels {
+namespace signal {
+namespace window {
+
+struct ExtractWindowsArgs {
+  int64_t window_length = -1;
+  int64_t window_step = -1;
+  int64_t window_center = -1;
+  int     axis = -1;
+  bool    reflect_pad = false;
+
+  inline bool operator==(const ExtractWindowsArgs& oth) const {
```

Comment:
Q: Is it used anywhere? E.g. in spectrogram args or sth?

Reply:
not used. It is here because I copied the original args struct from the FFT kernel (and there I am using the equality operator). That being said, it doesn't hurt so I am keeping it